### PR TITLE
explicitly install dependencies

### PR DESCRIPTION
**Pull request by chenzhiwei**
When building the Docker image on IBM Power system, the three dependencies must be explicitly installed.

Even I installed these dependencies, I still encounter following issue: `configurable-http-proxy not found`

```
[I 2018-12-18 07:25:02.812 JupyterHub app:1673] Using Authenticator: dummyauthenticator.dummyauthenticator.DummyAuthenticator
[I 2018-12-18 07:25:02.813 JupyterHub app:1673] Using Spawner: spawner.KubeFormSpawner
[I 2018-12-18 07:25:02.825 JupyterHub app:1055] Writing cookie_secret to /srv/jupyterhub/jupyterhub_cookie_secret
[I 2018-12-18 07:25:02.850 alembic.runtime.migration migration:117] Context impl SQLiteImpl.
[I 2018-12-18 07:25:02.851 alembic.runtime.migration migration:122] Will assume non-transactional DDL.
[I 2018-12-18 07:25:02.875 alembic.runtime.migration migration:327] Running stamp_revision  -> 896818069c98
[I 2018-12-18 07:25:03.229 JupyterHub proxy:431] Generating new CONFIGPROXY_AUTH_TOKEN
[W 2018-12-18 07:25:03.230 JupyterHub app:1173] No admin users, admin interface will be unavailable.
[W 2018-12-18 07:25:03.230 JupyterHub app:1174] Add any administrative users to `c.Authenticator.admin_users` in config.
[I 2018-12-18 07:25:03.230 JupyterHub app:1201] Not using whitelist. Any authenticated user will be allowed.
[I 2018-12-18 07:25:03.287 JupyterHub app:1855] Hub API listening on http://0.0.0.0:8081/hub/
[I 2018-12-18 07:25:03.287 JupyterHub app:1857] Private Hub API connect url http://jupyter-0:8081/hub/
[W 2018-12-18 07:25:03.289 JupyterHub proxy:565] Running JupyterHub without SSL.  I hope there is SSL termination happening somewhere else...
[I 2018-12-18 07:25:03.289 JupyterHub proxy:567] Starting proxy @ http://0.0.0.0:8000/
[E 2018-12-18 07:25:03.294 JupyterHub proxy:579] Failed to find proxy ['configurable-http-proxy']
    The proxy can be installed with `npm install -g configurable-http-proxy`.To install `npm`, install nodejs which includes `npm`.If you see an `EACCES` error or permissions error, refer to the `npm` documentation on How To Prevent Permissions Errors.
```

### CODE REVIEW

**Comment by minrk** [comment]
configurable-http-proxy is not used by the hub image in this chart. how are you building your images and creating jupyterhub_config.py?

**Comment by chenzhiwei** [comment]
kubeflow project uses this image in the ks files, I run `ks apply default -c jupyter` to enable the jupyter: https://github.com/kubeflow/kubeflow/blob/master/kubeflow/jupyter/jupyter.libsonnet#L150

**Comment by minrk** [comment]
Does kubeflow use the image for the hub from this chart outside this chart? How do they start the configurable-http-proxy service?

**Comment by chenzhiwei** [comment]
Kubeflow uses this outside this chart, it uses ksonnet to deploy jupyterhub-k8s, the command is:

https://github.com/kubeflow/kubeflow/blob/master/kubeflow/jupyter/jupyter.libsonnet#L145-L149

```
command: [
  "jupyterhub",
   "-f",
   "/etc/config/jupyter_config.py",
],
```

This PR does not include anything related to `configurable-http-proxy`, could you review and approve this PR?

File: images/hub/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+jhub_remote_user_authenticator==0.1.0
```
Comment (minrk) [inline images/hub/requirements.txt]: This PR adds an additional authenticator that can't be used with this chart. Can you explain why this is added here without any support for using it?

**Comment by minrk** [comment]
If kubeflow's not using this chart and it's not running a CHP container and it's not using the configuration from this image, it probably shouldn't be using the image that does all those things. Almost everything this image does is create a config file to load config from this chart's values.yaml. If that's not being used, there isn't much reason to use this image. I would suggest kubeflow build its own image for jupyterhub with its needed packages. Or am I missing something? What does kubeflow get from using this image instead of controlling their own?

Here's a minimal jupyterhub dockerfile:

```
# requirements.txt
jupyterhub==0.9.4
jupyterhub-kubespawner==0.10.1
# authenticators
```

```docker
FROM python:3.7
COPY requirements.txt /tmp/requirements.txt
RUN pip install --no-cache -r /tmp/requirements.txt
RUN mkdir /etc/jupyterhub && mkdir /srv/jupyterhub
COPY jupyterhub_config.py /etc/jupyterhub/jupyterhub_config.py
WORKDIR /srv/jupyterhub
CMD ["jupyterhub", "-f", "/etc/jupyerhub/jupyterhub_config.py"]
```

that's probably a lot easier to maintain than extending this image that makes a lot of assumptions that aren't satisfied by kubeflow.

**Comment by consideRatio** [comment]
@chenzhiwei while I have not learned enough about Kubeflow yet to navigate this fully, I'm closing this PR for now based on the following observations.

- it seems like Kubeflow needs different things from the docker image with JupyterHub than this helm Helm chart to deploy JupyterHub
- the provided config file for jupyterhub is very tightly coupled with this specific deployment
- the PR is stale and I know minrk isn't available for a while

**Pull request closed without merging**